jstests/oidc/oidc_command_list_keys.js

```diff
@@ -0,0 +1,77 @@
+import { OIDCFixture } from 'jstests/oidc/lib/oidc_fixture.js';
+
+const issuer_url1 = OIDCFixture.allocate_issuer_url();
+const issuer_url2 = OIDCFixture.allocate_issuer_url();
+const issuer_url3 = OIDCFixture.allocate_issuer_url();
+
+const oidcProviders = [
+    {
+        issuer: issuer_url1,
+        clientId: "clientId",
+        audience: "audience",
+        matchPattern: "1$",
+        authNamePrefix: "test",
+        useAuthorizationClaim: false,
+    },
+    {
+        issuer: issuer_url2,
+        clientId: "clientId",
+        audience: "audience",
+        matchPattern: "2$",
+        authNamePrefix: "test",
+        useAuthorizationClaim: false,
+    },
+    {
+        issuer: issuer_url3,
+        clientId: "clientId",
+        audience: "audience",
+        matchPattern: "3$",
+        authNamePrefix: "test",
+        useAuthorizationClaim: false,
+    },
+];
+
+const idps = [
+    {
+        url: issuer_url1,
+        config: {
+            number_of_jwks: 3,
+        }
+    },
+    {
+        url: issuer_url2,
+        config: {
+            number_of_jwks: 2,
+        }
+    },
+    {
+        url: issuer_url3,
+        config: {
+            number_of_jwks: 1,
+        }
+    },
+];
+
+var test = new OIDCFixture({ oidcProviders: oidcProviders, idps: idps });
+test.setup();
+
+// verifies the key sets for a given issuer_url and the expected number of keys
+const verifyKeys = (keys, issuer_url, expectedNumberOfKeys) => {
+    assert(keys.keySets, "oidcListKeys should have keySets");
+    assert(keys.keySets[issuer_url], "oidcListKeys should have keySets for issuer_url");
+    assert(keys.keySets[issuer_url].keys, "key set for issuer_url should have keys");
+    assert.eq(keys.keySets[issuer_url].keys.length, expectedNumberOfKeys, `key set for ${issuer_url} should have ${expectedNumberOfKeys} keys`);
```

Comment:
Please consider reducing the chain of assertions to a single line with conditional property access:
```suggestion
    assert.eq(keys?.keySets[issuer_url]?.keys?.length, expectedNumberOfKeys, `key set for ${issuer_url} should have ${expectedNumberOfKeys} keys`);
```
The similar change can be done in the `oidc_command_refresh_keys.js` test.